crates/biome_graphql_parser/src/parser/definitions/operation.rs

```diff
@@ -69,69 +62,213 @@ impl ParseRecovery for SelectionListParseRecovery {
+    }
+
+    // we don't need diagnostic here, because name is optional
+    parse_name(p).ok();
+    // we don't need diagnostic here, because variable definitions are optional
+    parse_variable_definitions(p).ok();
+
+    DirectiveList.parse_list(p);
+    parse_selection_set(p).ok();
+
+    Present(m.complete(p, GRAPHQL_OPERATION_DEFINITION))
+}
+
+#[inline]
+fn parse_selection_set(p: &mut GraphqlParser) -> ParsedSyntax {
+    let m = p.start();
+    p.expect(T!['{']);
+    SelectionList.parse_list(p);
+    p.expect(T!['}']);
+    Present(m.complete(p, GRAPHQL_SELECTION_SET))
+}
+
 #[inline]
 fn parse_selection(p: &mut GraphqlParser) -> ParsedSyntax {
-    // TODO: parse any selection
-    match p.cur() {
-        DOT3 => todo!(),
-        _ if is_at_field(p) => parse_field(p),
-        _ => Absent,
+    if is_at_field(p) {
+        parse_field(p)
+    } else if is_at_fragment(p) {
+        parse_fragment(p)
+    } else {
+        Absent
     }
 }
 
 #[inline]
 fn parse_field(p: &mut GraphqlParser) -> ParsedSyntax {
-    // TODO: parse alias, arguments, nested selection set
+    if !is_at_field(p) {
+        return Absent;
+    }
     let m = p.start();
-    parse_name(p).or_add_diagnostic(p, expected_name);
+    let next_token = p.lookahead();
+    if next_token == T![:] {
+        let m = p.start();
+        parse_name(p).ok();
+        p.bump(T![:]);
+        m.complete(p, GRAPHQL_ALIAS);
+    }
+    parse_name(p).ok();
+
+    // arguments are optional
+    parse_arguments(p).ok();
     DirectiveList.parse_list(p);
+
+    if is_at_selection_set(p) {
+        parse_selection_set(p).ok();
+    }
     Present(m.complete(p, GRAPHQL_FIELD))
 }
 
 #[inline]
-pub(crate) fn is_at_selection_set(p: &mut GraphqlParser<'_>) -> bool {
+fn parse_fragment(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !is_at_fragment(p) {
+        return Absent;
+    }
+    let m = p.start();
+    p.bump(DOT3);
+    if is_at_name(p) {
+        parse_name(p).ok();
+        DirectiveList.parse_list(p);
+        Present(m.complete(p, GRAPHQL_FRAGMENT_SPREAD))
+    } else {
+        if p.at(T![on]) {
+            let m = p.start();
+            p.bump(T![on]);
+            parse_named_type(p).or_add_diagnostic(p, expected_named_type);
+            m.complete(p, GRAPHQL_TYPE_CONDITION);
+        }
+        DirectiveList.parse_list(p);
+        parse_selection_set(p).ok();
+        Present(m.complete(p, GRAPHQL_INLINE_FRAGMENT))
+    }
+}
+
+#[inline]
+fn parse_variable_definitions(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !p.at(T!['(']) {
+        return Absent;
+    }
+
+    let m = p.start();
+
+    p.bump(T!['(']);
+    VariableDefinitionList.parse_list(p);
+    p.expect(T![')']);
+
+    Present(m.complete(p, GRAPHQL_VARIABLE_DEFINITIONS))
+}
+
+#[inline]
+fn parse_variable_definition(p: &mut GraphqlParser) -> ParsedSyntax {
+    if !is_at_variable(p) {
+        return Absent;
+    }
+
+    let m = p.start();
+
+    parse_variable(p).ok();
```

Comment:
Probably the error is handled downstream. It is still worth leaving a comment to say that